example/cajita_tutorial/11_interpolation/interpolation_example.cpp

```diff
@@ -0,0 +1,224 @@
+            double x[2];
+            local_mesh.coordinates( Cajita::Cell(), idx, x );
+            points( pid, Cajita::Dim::I ) = x[Cajita::Dim::I];
+            points( pid, Cajita::Dim::J ) = x[Cajita::Dim::J];
+        } );
+
+    // Next, we use Cajita functionality to create grid data fields.
+    // Create a scalar field on the grid. See the Array tutorial example for
+    // information on these functions.
+    auto scalar_layout =
+        Cajita::createArrayLayout( local_grid, 1, Cajita::Node() );
+    auto scalar_grid_field = Cajita::createArray<double, ExecutionSpace>(
+        "scalar_grid_field", scalar_layout );
+
+    // Create a halo for scatter operations. This concept is discussed in more
+    // detail in the Halo tutorial example.
+    auto scalar_halo =
+        Cajita::createHalo( *scalar_grid_field, Cajita::NodeHaloPattern<2>() );
+    auto scalar_grid_host =
+        Kokkos::create_mirror_view( scalar_grid_field->view() );
+
+    // Create a vector field on the grid.
+    auto vector_layout =
+        Cajita::createArrayLayout( local_grid, 2, Cajita::Node() );
+    auto vector_grid_field = Cajita::createArray<double, ExecutionSpace>(
+        "vector_grid_field", vector_layout );
+    auto vector_halo =
+        Cajita::createHalo( *vector_grid_field, Cajita::NodeHaloPattern<2>() );
+    auto vector_grid_host =
+        Kokkos::create_mirror_view( vector_grid_field->view() );
+
+    // Simple Kokkos::Views may be used to represent particle data.
+    // Create a scalar point field.
+    Kokkos::View<double*, ExecutionSpace> scalar_point_field(
+        Kokkos::ViewAllocateWithoutInitializing( "scalar_point_field" ),
+        num_point );
+    auto scalar_point_host = Kokkos::create_mirror_view( scalar_point_field );
+
+    // Create a vector point field.
+    Kokkos::View<double* [2], ExecutionSpace> vector_point_field(
+        Kokkos::ViewAllocateWithoutInitializing( "vector_point_field" ),
+        num_point );
+    auto vector_point_host = Kokkos::create_mirror_view( vector_point_field );
+
+    // Create a tensor point field.
+    Kokkos::View<double* [2][2], ExecutionSpace> tensor_point_field(
+        Kokkos::ViewAllocateWithoutInitializing( "tensor_point_field" ),
+        num_point );
+    auto tensor_point_host = Kokkos::create_mirror_view( tensor_point_field );
+
+    /***************************************************************************
+     * P2G
+     **************************************************************************/
+    /*
+     * The Cajita::P2G namespace contains several methods for interpolating data
+     * from particles to the grid. These interpolations are inherently scatter
+     * operations for particle-based threading (a single particle maps to
+     * several grid nodes), which requires an underlying Kokkos::ScatterView for
+     * the data being interpolated. Of note, these methods perform
+     * interpolations for a single particle datum. They may ...
```

Comment:
incomplete paragraph

Reply:
We need to add examples for the thread-level interface per @streeve comment as this will be most used. I think it makes sense to do it in this file to show how kernel fusion works. I also like the g2p2g example idea as a more advanced extension of individual operators. Also show how you can mix different spline data types (e.g. cell/node or quadratic/cubic) in a single kernel.

Reply:
So how about one additional example here of the per-thread interface and a new issue for creating `interpolation_advanced_fused` for g2p2g?